src/code/get_ensemble.r

```diff
@@ -0,0 +1,76 @@
+# R script to create ensemble forecats using models submitted to the CovidHub
+
+ref_date <- lubridate::ceiling_date(Sys.Date(), "week") - lubridate::days(1)
+hub_path <- "."
+task_id_cols <- c(
+  "reference_date", "location", "horizon",
+  "target", "target_end_date"
+)
+output_dirpath <- "CovidHub-ensemble/"
+if (!dir.exists(output_dirpath)) {
+  dir.create(output_dirpath, recursive = TRUE)
+}
+
+# Get current forecasts from the hub, excluding baseline and ensembles
+hub_content <- hubData::connect_hub(hub_path)
+current_forecasts <- hub_content |>
+  dplyr::filter(
+    reference_date == ref_date,
+    !str_detect(model_id, "CovidHub")
+  ) |>
+  hubData::collect_hub()
+
+yml_files <- list.files(paste0(hub_path, "/model-metadata"),
+  pattern = "\\.ya?ml$", full.names = TRUE
+)
+
+# Read model metadata and extract designated models
+designated_models <- purrr::map_chr(yml_files, function(file) {
+  yml_data <- yaml::yaml.load_file(file)
+  as.character(
+    ifelse(
+      "designated_model" %in% names(yml_data), yml_data$designated_model, NA
+    )
+  )
+})
+
+eligible_models <- data.frame(
+  Model = tools::file_path_sans_ext(basename(yml_files)),
+  Designated_Model = designated_models
+) |> dplyr::filter(Designated_Model == TRUE)
```

Comment:
This section can be simplified and made more robust and readable, I think:

```suggestion
 is_model_designated <- function(yaml_file) {
  yml_data <- yaml::yaml.load_file(yaml_file)
  team_and_model <- glue::glue("{yml_data$team_abbr}-{yml_data$model_abbr}") 
  is_designated <- ifelse("designated_model" %in% names(yml_data), 
                          as.logical(yml_data$designated_model), 
                          FALSE)
  return(list(Model = team_and_model, Designated_Model = is_designated))
}

eligible_models <- purrr::map(yml_files, is_model_designated) |> 
   dplyr::bind_rows() |> 
   dplyr::filter(Designated_Model)
```

Reply:
changed in https://github.com/CDCgov/covid19-forecast-hub/pull/24/commits/656f5ed19952847067eb17925885bb0274c8fc4f